### DISCUSSION TITLE
Cannot find module 'react-query/types/core/utils'

### DISCUSSION THREAD

**Discussion by vera-barysava** [TanStack/query#3257] [Q&A]
Hello!

There is a problem with importing Updater from react-query/types/core/utils.'

`import { Updater } from 'react-query/types/core/utils';`

During project building, the error Cannot find module 'react-query/types/core/utils appears.
This started after updating the version of react-query from 3.18.1 to 3.34.12.
The current version of TypeScript is 3.8.3.

Are there some major updates with importing that were released after react-query 3.18.1?

**Comment by TkDodo** [comment]
please don't import from "internals" because we often shift them around. As with any library, it is best to stick to the publicly exported interface, so:

```
import { Updater } from 'react-query';
```

if that doesn't exist, we don't export it.

**Reply by llllvvuu** [reply]
This type can be imported from `@tanstack/query-core` but is not being re-exported by `react-query`. That would require:
```
diff --git a/packages/react-query/src/index.ts b/packages/react-query/src/index.ts
index b13a86a6..de2db7ff 100644
--- a/packages/react-query/src/index.ts
+++ b/packages/react-query/src/index.ts
@@ -5,6 +5,7 @@ import './setBatchUpdatesFn'
 
 // Re-export core
 export * from '@tanstack/query-core'
+export type * from '@tanstack/query-core'
 
 // React Query
 export * from './types'

```

But `export type * from` is only supported in TypeScript 5: https://github.com/microsoft/TypeScript/pull/52217 . It looks like [`react-query` is on TypeScript 4.7.4](https://github.com/TanStack/query/blob/main/package.json#L106).

**Reply by TkDodo** [reply]
so `export * from '@tanstack/query-core'` is not re-exporting types? How come I can then import types that are in the core, like `QueryFilters` ?

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQFA2iSxwwCeYWcA3nAIoCuWKCwBiwADYxB6AL5wiJcgAEYyAHaoVuANYB6SnhgBaAI4ChtGrgjr4ADwBcvMyPGSo6ALxcacOKcEA0lgsjjBQAjTSQA

**Reply by llllvvuu** [reply]
Ah, this is my bad, I was using v3 apparently by importing from `react-query` instead of `@tanstack/react-query`. It is working in v4.
